src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1482 @@
+        unsafe {
+            let mut server_pid: winapi::ULONG = 0;
+            if kernel32::GetNamedPipeServerProcessId(*self.handle, &mut server_pid) == winapi::FALSE {
+                return Err(WinError::last("GetNamedPipeServerProcessId"));
+            }
+            Ok(server_pid)
+        }
+    }
+
+    // TODO(vlad): if we could guarantee that a server handle can't be passed
+    // after it's already been used to start receiving data, we could
+    // store the server handle instead of needing to look it up each time.
+    fn get_pipe_server_process_handle(&self) -> Result<WinHandle,WinError> {
+        unsafe {
+            let server_pid = try!(self.get_pipe_server_process_id());
+            if server_pid == *CURRENT_PROCESS_ID {
+                return Ok(WinHandle::new(*CURRENT_PROCESS_HANDLE as HANDLE));
+            }
+
+            let raw_handle = kernel32::OpenProcess(winapi::PROCESS_DUP_HANDLE,
+                                                   winapi::FALSE,
+                                                   server_pid as winapi::DWORD);
+            if raw_handle.is_null() {
+                return Err(WinError::last("OpenProcess"));
+            }
+
+            Ok(WinHandle::new(raw_handle))
+        }
+    }
+
+    fn maximum_message_data_size(oob_data_len: u64) -> usize {
+        assert!((oob_data_len as usize) < (WRITE_BUFFER_SIZE-HEADER_SIZE), "too much oob data");
+        (WRITE_BUFFER_SIZE-HEADER_SIZE) - (oob_data_len as usize)
+    }
+
+    fn needs_fragmentation(data_len: usize, oob: &OsIpcOutOfBandMessage) -> bool {
+        let oob_size = if oob.needs_to_be_sent() { bincode::serde::serialized_size(oob) } else { 0 };
+        data_len > OsIpcSender::maximum_message_data_size(oob_size)
+    }
+
+    // An internal-use-only send method that sends just data, but doesn't
+    // fragment -- we use this for the secondary channel fragmentation
+    // creates.
+    fn send_unfragmented(&self, data: &[u8]) -> Result<(),WinError> {
+        assert!(data.len() < INVALID_HEADER_DATA_SIZE as usize);
+
+        dd2!("[c {:?}] writing unfragmented to (pid {}->{})", *self.handle, *CURRENT_PROCESS_ID,
+             try!(self.get_pipe_server_process_id()));
+
+        unsafe {
+            let header: [u32; 2] = [ data.len() as u32, 0 ];
+            let header_bytes: &[u8] = slice::from_raw_parts(header.as_ptr() as *const u8, HEADER_SIZE);
+            try!(write_buf(*self.handle, &header_bytes));
+            try!(write_buf(*self.handle, data));
+        }
+        dd2!("[c {:?}] sent unfragmented: {} bytes", *self.handle, data.len());
+        Ok(())
+    }
+
+    pub fn send(&self,
+                data: &[u8],
+                ports: Vec<OsIpcChannel>,
+                shared_memory_regions: Vec<OsIpcSharedMemory>)
+                -> Result<(),WinError>
+    {
+        // We limit the max size we can send here; we can fix this
+        // just by upping the header to be 2x u64 if we really want
+        // to.
```

Comment:
Since it's not entirely clear what is best here, I don't mind that you didn't change it -- however, I think it would be good to bring this up in the comment, so the consideration doesn't get lost?...